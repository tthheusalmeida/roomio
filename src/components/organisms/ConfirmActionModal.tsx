"use client";

import Button from "@/components/atoms/Button";
import Modal from "@/components/atoms/Modal";

interface ConfirmActionProps {
  isOpen: boolean;
  expanded?: boolean;
  title?: string;
  message?: React.ReactNode | string;
  onConfirm?: (confirm: boolean) => void;
  onCancel?: (confirm: boolean) => void;
}

export default function ConfirmActionModal({
  isOpen = false,
  expanded = false,
  title = "",
  message = "",
  onConfirm,
  onCancel,
}: ConfirmActionProps) {
  const handleOnConfirm = () => {
    if (onConfirm) {
      onConfirm(true);
    }
  };

  const handleCancel = () => {
    if (onCancel) {
      onCancel(false);
    }
  };

  return (
    <Modal isOpen={isOpen} expanded={expanded} title={title}>
      <div className={`${expanded ? "h-screen -my-4 py-4" : ""} flex flex-col`}>
        <div className="flex flex-col justify-center gap-4 mb-4">
          <span className="text-gray-700 text-sm transition-opacity duration-600 min-h-14 opacity-100">
            {message}
          </span>
        </div>

        <div className="flex gap-4">
          <Button variant="secondary" onClick={handleCancel} isFull={true}>
            Cancelar
          </Button>

          <Button variant="primary" onClick={handleOnConfirm} isFull={true}>
            Confirmar
          </Button>
        </div>
      </div>
    </Modal>
  );
}
