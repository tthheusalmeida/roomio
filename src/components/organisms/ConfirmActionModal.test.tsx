import { render, screen, fireEvent } from "@testing-library/react";
import ConfirmActionModal from "./ConfirmActionModal";

jest.mock("@/components/molecules/Modal", () => ({ isOpen, children }: any) => {
  return isOpen ? <div data-testid="modal">{children}</div> : null;
});

jest.mock("@/components/atoms/Button", () => ({ children, onClick }: any) => {
  return <button onClick={onClick}>{children}</button>;
});

describe("ConfirmActionModal component", () => {
  it("renders correctly when open", () => {
    render(
      <ConfirmActionModal
        isOpen={true}
        title="Test Title"
        message="Test Message"
      />
    );

    expect(screen.getByTestId("modal")).toBeInTheDocument();
    expect(screen.getByText("Test Message")).toBeInTheDocument();
    expect(screen.getByText("Cancelar")).toBeInTheDocument();
    expect(screen.getByText("Confirmar")).toBeInTheDocument();
  });

  it("does not render when closed", () => {
    render(<ConfirmActionModal isOpen={false} />);
    expect(screen.queryByTestId("modal")).not.toBeInTheDocument();
  });

  it("calls onConfirm when Confirmar button is clicked", () => {
    const onConfirm = jest.fn();
    render(<ConfirmActionModal isOpen={true} onConfirm={onConfirm} />);

    fireEvent.click(screen.getByText("Confirmar"));
    expect(onConfirm).toHaveBeenCalledWith(true);
  });

  it("calls onCancel when Cancelar button is clicked", () => {
    const onCancel = jest.fn();
    render(<ConfirmActionModal isOpen={true} onCancel={onCancel} />);

    fireEvent.click(screen.getByText("Cancelar"));
    expect(onCancel).toHaveBeenCalledWith(false);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <ConfirmActionModal
        isOpen={true}
        expanded={true}
        title="Snapshot Test"
        message="Snapshot message"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
