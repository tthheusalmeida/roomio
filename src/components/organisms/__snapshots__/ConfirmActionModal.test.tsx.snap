// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ConfirmActionModal component matches snapshot 1`] = `
<div>
  <div
    data-testid="modal"
  >
    <div
      class="h-screen -my-4 py-4 flex flex-col"
    >
      <div
        class="flex flex-col justify-center gap-4 mb-4"
      >
        <span
          class="text-gray-700 text-sm transition-opacity duration-600 min-h-14 opacity-100"
        >
          Snapshot message
        </span>
      </div>
      <div
        class="flex gap-4"
      >
        <button>
          Cancelar
        </button>
        <button>
          Confirmar
        </button>
      </div>
    </div>
  </div>
</div>
`;
